modin/pandas/test/test_dataframe.py

```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
-                        )
-                else:
-                    modin_multi_level_result = modin_df_multi_level.all(
-                        axis=axis, bool_only=bool_only, level=level, skipna=skipna
-                    )
-
-                    df_equals(modin_multi_level_result, pandas_multi_level_result)
-
-    @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-    @pytest.mark.parametrize("axis", axis_values, ids=axis_keys)
     @pytest.mark.parametrize(
         "skipna", bool_arg_values, ids=arg_keys("skipna", bool_arg_keys)
     )
     @pytest.mark.parametrize(
         "bool_only", bool_arg_values, ids=arg_keys("bool_only", bool_arg_keys)
     )
-    def test_any(self, data, axis, skipna, bool_only):
-        modin_df = pd.DataFrame(data)
-        pandas_df = pandas.DataFrame(data)
-
-        try:
-            pandas_result = pandas_df.any(axis=axis, skipna=skipna, bool_only=bool_only)
-        except Exception as e:
-            with pytest.raises(type(e)):
-                modin_df.any(axis=axis, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.any(axis=axis, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = pandas_df.any(axis=None, skipna=skipna, bool_only=bool_only)
-        except Exception as e:
-            with pytest.raises(type(e)):
-                modin_df.any(axis=None, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.any(axis=None, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
+    def test_all_any(self, skipna, bool_only):
+        data = {
+            "int_col": [1, 2, np.NaN, 4],
+            "float_col": [np.NaN, 9.4, 10.1, np.NaN],
+            "str_col": ["a", np.NaN, "c", "d"],
+            "bool_col": [False, True, np.NaN, np.NaN],
+        }
+        modin_df, pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
 
-        try:
-            pandas_result = pandas_df.T.any(
-                axis=axis, skipna=skipna, bool_only=bool_only
+        for method in ["all", "any"]:
+            eval_general(
+                modin_df,
+                pandas_df,
+                lambda df: getattr(df, method)(skipna=skipna, bool_only=bool_only),
             )
-        except Exception as e:
-            with pytest.raises(type(e)):
-                modin_df.T.any(axis=axis, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.T.any(axis=axis, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = pandas_df.T.any(
-                axis=None, skipna=skipna, bool_only=bool_only
+            # transposed
+            eval_general(
+                modin_df,
+                pandas_df,
+                lambda df: getattr(df.T, method)(skipna=skipna, bool_only=bool_only),
```

Comment:
I think we also can move transpose-or-not-transpose variable to a function parameter, to explicitly see in a pytest error message which case is failed